pmd-plsql/etc/grammar/PldocAST.jjt

```diff
@@ -4556,13 +4644,75 @@ SKIP :
 	" " | "\t" | "\f"
 }
 
+/* 
+This Grammar (or JavaCC) has several bugs.
+
+Currently (2021-02-26) it fails for things like
+- dbms_lob.trim(...),
+- scalar subqueries in the WHERE-clause,
+- TREAT(object AS type),
+just to name a few.
+
+Running PMD over a code base of approx. 1100 DDL scripts
+from a commercial product, the parser failed for 182 files.
+About 10% of these parsing errors actually showed flaws in the code,
+e.g. "\u2013" instead of "-", or utf-8 encoding instead of windows-1252
```

Comment:
Would you mind to give an example of the encoding problems you have seen? Ideally you'll create a new issue at https://github.com/pmd/pmd/issues with a minimal example sql code snippet.

Reply:
The encoding problems were in fact encoding problems, so not really related to PMD:
- Some files where the actual encoding didn't match our NLS_LANG setting (WE8MSWIN1252) and the encoding I specified on the pmd command line
- Some files which used unusual characters.

What I wanted to say is:
If PMD cannot parse a file, this can have different reasons:
a) Bugs/flaws in the script file itself
b) Encoding problems (wrong encoding or uncommon characters used)
c) Bugs in the PMD parser.
About 90% of the PMD parsing errors for our code base were caused by c).

BTW regarding b): The PL/SQL parser seems to be focused on German scripts. E.G. I find the umlauts ü,ö,ü mentioned in the PldocAST.jjt file. I think that the parser might fail if a script contains e.g. greek letters or accented characters apart from German umlauts/ szglig at certain places (e.g. as part of identifiers or SQL*Plus commands). But I didn't investigate this further, as it is not a problem for me :-)

Reply:
Yes, that focus on the German umlauts is weird. It seems to be extending the possible characters of identifiers. But there are of course more valid characters allowed.

> The database character set defines which characters are classified as letters and digits

https://docs.oracle.com/en/database/oracle/oracle-database/21/lnpls/plsql-language-fundamentals.html#GUID-321084DE-3BDD-484B-AE85-7A991B67C51A

And an identifier must start with a letter and consists of letters and digits...

That's probably for some other pull request to clean this up.